test/expression/eval_test.exs

```diff
@@ -113,6 +113,16 @@ defmodule Expression.EvalTest do
              })
   end
 
+  test "if with nil values" do
+    {:ok, ast, "", _, _, _} =
+      Parser.parse("@if(status,\nLEFT(status, 2),\nfalse)")
+
+    assert false ==
```

Comment:
Without this change, this AST will break because it will try to "LEFT" "nil" but this is not needed because if will evaluate to false and so return it

Reply:
Thanks for this contribution! I was looking into a similar problem.
Could we extend this solution to `and`/`or` boolean evaluations? I've been trying a lazy evaluation of `and` with a `nil` value, but it seems to be failing. Here’s an example test:

```
test "and with nil values" do
  {:ok, ast, "", _, _, _} = Parser.parse("@and(isstring(status), LEFT(status, 2))")

  assert false == Eval.eval!(ast, %{ "status" => nil})
end
```

Reply:
Hi @Djack1010, I don't think there is a good way to fix the "and" because unfortunately every argument of the function needs to be evaluated and not return error in order to check if the function can return true or false. This fix would only work for OR and IFs